### DISCUSSION TITLE
Errors after moving ethers from v5 to v6

### DISCUSSION THREAD

**Discussion by D1scoDancer** [wevm/wagmi#2190] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/60194526/232436078-e6702cbf-ffe2-4847-be3d-7e0ff0ab160a.png)
these and several others errors related to new package layout of ethers@6.*.*

Currently I have the atest versions of wagmi, ethers, connectkit 
![image](https://user-images.githubusercontent.com/60194526/232436347-30acfe1a-4291-448e-85a6-6dbfc757d700.png)

Any ideas how to solve it? Or maybe this mistake is related to something else?

**Comment by D1scoDancer** [comment]
I guess wagmi doesn't support ethers@6 yet. Any hope it will happen in the near future? Or I should just return to ethers@5?

**Reply by jxom** [reply, marked as answer]
wagmi will not be migrating to ethers v6. We will be migrating to [viem](https://viem.sh) for wagmi v1 to achieve better stability, type safety, lower bundle size and performance.

**Reply by D1scoDancer** [reply]
got it, thanks

**Reply by firnprotocol** [reply]
related question: how long before v1, together with viem v1, are released? i am ready to migrate web3react --> wagmi, but i would prefer to migrate ethers --> viem simultaneously.

are we talking a month? a year? please let me know; thanks.

**Reply by chybisov** [reply]
@jxom please any ETA for [viem](https://viem.sh/) migration 🙏 Thank you!

**Reply by jxom** [reply]
It'll be a few weeks. Going to post a GitHub discussion on v1 Roadmap tomorrow (or Thursday)!
